internal/backends/winit/lib.rs

```diff
@@ -233,7 +234,10 @@ impl i_slint_core::platform::Platform for Backend {
     }
 
     fn run_event_loop(&self) -> Result<(), PlatformError> {
-        crate::event_loop::run()
+        let loop_state = self.event_loop_state.borrow_mut().take().unwrap_or_default();
+        let (new_state, res) = loop_state.run();
```

Comment:
should run perhaps take a &mut self?
So you can do
```rust
self.event_loop_state.borrow_mut().get_or_insert_with(Default::default).run()
```

Reply:
I had the same thought, but it doesn't compile with the wasm build (that uses run() instead of run_return()):

```
error[E0521]: borrowed data escapes outside of method
   --> internal/backends/winit/event_loop.rs:701:13
    |
644 |       pub fn run(&mut self) -> Result<(), corelib::platform::PlatformError> {
    |                  ---------
    |                  |
    |                  `self` is a reference that is only valid in the method body
    |                  let's call the lifetime of this reference `'1`
...
701 | /             winit_loop.run(
702 | |                 move |event: Event<SlintUserEvent>,
703 | |                       event_loop_target: &EventLoopWindowTarget<SlintUserEvent>,
704 | |                       control_flow: &mut ControlFlow| {
...   |
712 | |                 },
713 | |             )
    | |             ^
    | |             |
    | |_____________`self` escapes the method body here
    |               argument requires that `'1` must outlive `'static`
    |
    = note: requirement occurs because of the type `RefCell<(Box<dyn ClipboardProvider>, Box<dyn ClipboardProvider>)>`, which makes the generic argument `(Box<dyn ClipboardProvider>, Box<dyn ClipboardProvider>)` invariant
    = note: the struct `RefCell<T>` is invariant over the parameter `T`
    = help: see <https://doc.rust-lang.org/nomicon/subtyping.html> for more information about variance
```